website/project/model.py

```diff
@@ -316,6 +352,12 @@ def edit(self, content, auth, save=False):
         self.content = content
         self.modified = True
         self.date_modified = datetime.datetime.utcnow()
+        new_mentions = get_valid_mentioned_users_guids(self, self.node.contributors)
+
+        if new_mentions:
+            project_signals.mention_added.send(self, new_mentions=new_mentions, auth=auth)
+            self.ever_mentioned.extend(new_mentions)
+
         if save:
```

Comment:
> Not sure if this will answer your question but awhile ago I polled the [dev flow](https://www.flowdock.com/app/cos/osf-dev?filter=search&query=can%20anyone%20explain%20why%20many) with a similar question.

Yes! That's the [thread](https://www.flowdock.com/app/cos/osf-dev/threads/2Ns3_QThVQFSkT2nk-HUuSTIkVm) I was thinking of. Thanks! :-)

I read that thread as being inconclusive. Skimming through the `self.save` calls in [`project/model.py`](https://github.com/leb2dg/osf.io/blob/6d3210ceead6e71d341f19f5399a226bd547a4e6/website/project/model.py) likewise suggests that we've got a somewhat rhapsodic approach overall. The challenge here is that if someone calls `edit` or `create` with `save=False`, and then calls `save`, they won't get the notifications or logging. They would need to reimplement that logic or know they want to in fact skip it.

However, given how messy saving is in our Project model—and `project/model.py` is clearly far overripe for a much larger refactor—I suppose our goal on this PR should be to rock that boat as little as possible.

> The `edit` case should be updated though so that `ever_mentioned` is only modified if `save`.

wfm